### PR TITLE
SOC-5600 : Use the new social avatar endpoints

### PR DESCRIPTION
**Pull request by MrHanachoo**
This PR changes users/spaces avatars urls in the platform:

user:
/rest/social/identity/organization/USER_ID/avatar  => /rest/v1/social/users/USER_ID/avatar

space:
/rest/social/identity/space/SPACE_ID/avatar  => /rest/v1/social/spaces/SPACE_ID/avatar

### CODE REVIEW

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/EntityConverterUtils.java [hunk 1]
```diff
@@ -66,9 +65,10 @@ public static void mapToProfile(IdentityEntity entity, Profile p) {
     
     Map<String, Object> props = p.getProperties();
     String providerId = entity.getProviderId();
+    Identity identity = p.getIdentity();
     if (!OrganizationIdentityProvider.NAME.equals(providerId) && !SpaceIdentityProvider.NAME.equals(providerId)) {
       p.setUrl(properties.get(Profile.URL));
-      p.setAvatarUrl(properties.get(Profile.AVATAR_URL));
+      p.setAvatarUrl(LinkProvider.buildAvatarURL(identity.getProviderId(), identity.getRemoteId()));
```
Comment (thomasdelhomenie) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/EntityConverterUtils.java]: I guess we can remove Profile.AVATAR_URL now ?

File: component/core/src/main/java/org/exoplatform/social/core/service/LinkProvider.java [hunk 2]
```diff
@@ -437,6 +442,39 @@ public static IdentityManager getIdentityManager() {
...
+    String username = remoteId;
+    
+    try {
+      username = URLEncoder.encode(username, "UTF-8");
+    } catch (UnsupportedEncodingException ex) {
+      LOG.warn("Failure to encode username for build URL", ex);
+    }
+    
+    if(providerId.equals(OrganizationIdentityProvider.NAME)) {
+      return new StringBuilder("/rest").append(BASE_URL).append("/users")
```
Comment (thomasdelhomenie) [inline component/core/src/main/java/org/exoplatform/social/core/service/LinkProvider.java]: "rest" should be replaced by CommonsUtils.getRestContextName().

File: component/core/src/main/java/org/exoplatform/social/core/service/LinkProvider.java [hunk 3]
```diff
@@ -68,6 +71,8 @@
    * Hacks for unit test to work.
    */
   public static String DEFAULT_PORTAL_OWNER = "classic";
+  
+  private static final  String BASE_URL = "/v1/social";
```
Comment (thomasdelhomenie) [inline component/core/src/main/java/org/exoplatform/social/core/service/LinkProvider.java]: Should be more meaningful, something like BASE_URL_SOCIAL_REST_API.

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/EntityConverterUtils.java (hunk 1 above)
Comment (MrHanachoo) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/EntityConverterUtils.java]: done

File: component/core/src/main/java/org/exoplatform/social/core/service/LinkProvider.java (hunk 3 above)
Comment (MrHanachoo) [inline component/core/src/main/java/org/exoplatform/social/core/service/LinkProvider.java]: done

File: component/core/src/main/java/org/exoplatform/social/core/service/LinkProvider.java (hunk 2 above)
Comment (MrHanachoo) [inline component/core/src/main/java/org/exoplatform/social/core/service/LinkProvider.java]: done

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSIdentityStorageImpl.java [hunk 4]
```diff
@@ -117,7 +117,7 @@ private void mapToProfileEntity(Profile profile, IdentityEntity entity) {
     String providerId = profile.getIdentity().getProviderId();
     if (!OrganizationIdentityProvider.NAME.equals(providerId) && !SpaceIdentityProvider.NAME.equals(providerId)) {
       entityProperties.put(Profile.URL, profile.getUrl());
-      entityProperties.put(Profile.AVATAR_URL, profile.getAvatarUrl());
+      //entityProperties.put(Profile.AVATAR_URL, profile.getAvatarUrl());
```
Comment (thomasdelhomenie) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSIdentityStorageImpl.java]: To be removed.

File: component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/storage/dao/ProfileDAOTest.java [hunk 5]
```diff
@@ -101,7 +101,7 @@ private IdentityEntity createProfile() {
     
     Map<String, String> props = new HashMap<String, String>();
     props.put(Profile.URL, "/profile/root");
-    props.put(Profile.AVATAR_URL, "/profile/root/avatar.png");
+    //props.put(Profile.AVATAR_URL, "/profile/root/avatar.png");
```
Comment (thomasdelhomenie) [inline component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/storage/dao/ProfileDAOTest.java]: To be removed.

Comment (MrHanachoo) [inline component/core-jpa/src/test/java/org/exoplatform/social/core/jpa/storage/dao/ProfileDAOTest.java]: done

File: component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSIdentityStorageImpl.java (hunk 4 above)
Comment (MrHanachoo) [inline component/core-jpa/src/main/java/org/exoplatform/social/core/jpa/storage/RDBMSIdentityStorageImpl.java]: done

**Pull request merged**